libdnf/utils/temp.cpp

```diff
@@ -27,22 +27,27 @@ along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
 
 #include <cstdlib>
 
+#include <stdio.h>
+#include <unistd.h>
+
 
 namespace libdnf::utils {
 
 
-TempDir::TempDir(const std::string & prefix) : TempDir(std::filesystem::temp_directory_path(), prefix) {}
+TempDir::TempDir(const std::string & name_prefix) : TempDir(std::filesystem::temp_directory_path(), name_prefix) {}
 
 
-TempDir::TempDir(const std::string & destdir, const std::string & prefix) {
-    std::filesystem::path dir = destdir;
-    dir /= prefix + "XXXXXX";
+TempDir::TempDir(std::filesystem::path destdir, const std::string & name_prefix) {
+    destdir /= name_prefix + ".XXXXXX";
 
-    const char * temp_path = mkdtemp(const_cast<char *>(dir.native().c_str()));
+    // mkdtemp modifies the dest's char * in-place
+    std::string dest = destdir;
+    const char * temp_path = mkdtemp(&*dest.begin());
```

Comment:
And using `front` (Returns reference to the first character in the string. ) is probably better than constructing an iterator.
`fd = mkstemp(&dest.front());`

Or since C++17 `data` (Returns a pointer to the underlying array serving as character storage.) can be used.
`fd = mkstemp(dest.data());`

Reply:
Right, `data()` is the better one, I missed that it returns a non-const pointer since C++17. Fixed.